### PR TITLE
Implement paginate and simplePaginate methods

### PR DESCRIPTION
**Pull request by vladyslavstartsev**
I've implemented paginate and simplePaginate methods for Collection like we have in Builder.

### CODE REVIEW

File: tests/PaginateTest.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+namespace Spatie\CollectionMacros\Test;
+
+use Illuminate\Support\Collection;
+
+class PaginateTest extends TestCase
+{
+    public function setUp()
+    {
+        $this->p = (new Collection(['item1', 'item2', 'item3', 'item4']))->paginate(2, 'page', 2, null);
```
Comment (freekmurze) [inline tests/PaginateTest.php]: Abbreviations are almost always a bad thing in PHP code. Could you rename `p` to something meaningful?

Comment (vladyslavstartsev) [inline tests/PaginateTest.php]: sure thing, I was just trying to implement similar tests that they have in `illuminate/pagination` package

Comment (freekmurze) [inline tests/PaginateTest.php]: Thanks, we'll take a more detailed look at the code in the coming days. But on first glance it looks pretty good.

File: src/macros.php [hunk 2]
```diff
@@ -297,3 +299,49 @@
...
+     * @param string $pageName
+     *
+     * @return \Illuminate\Pagination\LengthAwarePaginator
+     */
+    Collection::macro('paginate', function (int $perPage = 15, string $pageName = 'page', int $page = null, int $total = null, array $options = []): LengthAwarePaginator {
+        $page = $page ?: LengthAwarePaginator::resolveCurrentPage($pageName);
+        $results = $this->forPage($page, $perPage);
+        $total = $total ?: $this->count();
+        $options += [
+            'path' => LengthAwarePaginator::resolveCurrentPath(),
```
Comment (sebastiandedeyne) [inline src/macros.php]: Curious, why is this line necessary?

File: src/macros.php [hunk 3]
```diff
@@ -297,3 +299,49 @@
...
+     * Paginate the given collection
+     *
+     * @param int $perPage
+     * @param int $total
+     * @param int $page
+     * @param string $pageName
+     *
+     * @return \Illuminate\Pagination\LengthAwarePaginator
+     */
+    Collection::macro('paginate', function (int $perPage = 15, string $pageName = 'page', int $page = null, int $total = null, array $options = []): LengthAwarePaginator {
```
Comment (sebastiandedeyne) [inline src/macros.php]: Please exactly match the `LengthAwarePaginator`'s signature for the method.

`$items, $total, $perPage, $currentPage = null, array $options = []`

File: src/macros.php [hunk 4]
```diff
@@ -297,3 +299,49 @@
...
+    /*
+     * Paginate the collection into a simple paginator
+     *
+     * @param int $perPage
+     * @param int $page
+     * @param string $pageName
+     *
+     * @return \Illuminate\Contracts\Pagination\Paginator
+     */
+    Collection::macro('simplePaginate', function (int $perPage = 15, string $pageName = 'page', int $page = null, array $options = []): Paginator {
```
Comment (sebastiandedeyne) [inline src/macros.php]: Please exactly match the `Paginator's signature for the method.

`$items, $perPage, $currentPage = null, array $options = []`

File: src/macros.php (hunk 3 above)
Comment (vladyslavstartsev) [inline src/macros.php]: I was hesitating  how to do it, I thought it would be better if I do it like we have in `\Illuminate\Database\Eloquent\Builder::paginate()`
so for example you do `\App\Users::paginate(20)`, and for collections you can do `$yourCollection->pagiinate(20)`
The problem with my solution is that there is no columns in collection (but I can do that it will grab only certain "column" in collection) and options are not available.
Do you get what I mean?
Do I still need to do it your way?

File: src/macros.php (hunk 2 above)
Comment (vladyslavstartsev) [inline src/macros.php]: The `path` key? Because you can do it with other path not for your site. For example, you have website `foo.bar`  and you put `site.com` in `path` key, links method will generate you something like `site.com?page=1`, `site.com?page=2`, and etc. 
Anything else?

File: src/macros.php (hunk 4 above)
Comment (vladyslavstartsev) [inline src/macros.php]: as above

File: src/macros.php (hunk 3 above)
Comment (sebastiandedeyne) [inline src/macros.php]: Didn't realize the `Builder` used that parameter order. In that case, ignore my inital commend!

I also agree that columns aren't useful here. If people really need to selectively paginate data, they can `map` their collection first.

**Comment by sebastiandedeyne** [comment]
Released in the latest version, thanks!

File: tests/SimplePaginateTest.php [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        $this->assertEquals([2 => 'item3'], $p->items());
+        $pageInfo = [
+            'per_page' => 2,
+            'current_page' => 2,
+            'next_page_url' => null,
+            'prev_page_url' => '/?page=1',
+            'from' => 3,
+            'to' => 3,
+            'data' => [2 => 'item3'],
+            'path' => '/',
```
Comment (mdrost) [inline tests/SimplePaginateTest.php]: `Paginator` returns `path` since [version 5.4.24](https://github.com/laravel/framework/releases/tag/v5.4.24) so this test is broken with earlier versions.
